### PR TITLE
Define module for managing LDAP authentication for E-Series

### PR DESCRIPTION
**Pull request by lmprice**
This module will allow users to manage LDAP authentication domains for
accessing E-Series systems.

##### SUMMARY

This is a new module to allow an owner of an E-Series system to manage the LDAP authentication domains that can be utilized for logging into the system.

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

netapp_e_ldap
##### ANSIBLE VERSION

```
ansible 2.6.0 (detached HEAD ce8c336d52) last updated 2018/04/18 11:23:03 (GMT -500)
  config file = None
  configured module search path = [u'/home/lorenp/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/lorenp/ansible/lib/ansible
  executable location = /home/lorenp/ansible/bin/ansible
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]

```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

File: lib/ansible/modules/storage/netapp/netapp_e_ldap.py
Comment (hulquest) [inline lib/ansible/modules/storage/netapp/netapp_e_ldap.py:254]: Could this be moved to the common fragment?  I don't think it's a must-fix for this PR.

File: lib/ansible/module_utils/netapp.py
Comment (mattclay) [inline lib/ansible/module_utils/netapp.py]: Use this instead:

```python
from ansible.module_utils.ansible_release import __version__
```

File: lib/ansible/modules/storage/netapp/netapp_e_ldap.py
Comment (lmprice) [reply lib/ansible/modules/storage/netapp/netapp_e_ldap.py:254]: I think that's a good idea, but let's get this one merged first. I'll add the common code after all of the other changes to the common code in netapp utils file.

File: lib/ansible/module_utils/netapp.py
Comment (lmprice) [reply lib/ansible/module_utils/netapp.py]: Got it. I was guessing on the best place to pull that information.

**Comment by hulquest** [comment]
shipit

**Review by carchi8py — APPROVED** [review]
shipit

**Comment by ndswartz** [comment]
shipit

**Review by ndswartz — APPROVED** [review]

**Comment by lmprice** [comment]
@mattclay Is this waiting on a review from a repository owner, or is there another piece of the merge process I'm missing?

**Comment by mattclay** [comment]
@lmprice The module will be merged once a member of the Ansible organization has reviewed it and decided to include it.

See: https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md#new-modules

**Comment by mattclay** [comment]
The sanity failure is unrelated and has been fixed. I've restarted CI.

**Review by ndswartz — APPROVED** [review]
shipit

**Comment by gundalow** [comment]
Can you please rebase this by following http://docs.ansible.com/ansible/devel/dev_guide/developing_rebasing.html Thanks in advance

**Comment by lmprice** [comment]
This is an expected failure. Waiting on #44659 to merge.

File: lib/ansible/modules/storage/netapp/netapp_e_ldap.py
Comment (gundalow) [reply lib/ansible/modules/storage/netapp/netapp_e_ldap.py:254]: @lmprice Reminder that this may need moving to common code

**Pull request merged**
